jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -407,32 +379,14 @@ private static void addModuleToCodePath(@NotNull GeneralCommandLine commandLine,
     }
   }
 
-  private static void markDirtyNodes(@NotNull List<Node> dirtyModules,
-                                     @NotNull GraphGenerator<Node> graph) {
-    for (Node node : dirtyModules) {
-      if (node != null) {
-        markDirtyNodes(node, graph);
-      }
-    }
-  }
-
-  private static void markDirtyNodes(@NotNull Node node, @NotNull GraphGenerator<Node> graph) {
-    if (node.myDirty) return;
-    node.myDirty = true;
-    Iterator<Node> childIterator = graph.getOut(node);
-    while (childIterator.hasNext()) {
-      markDirtyNodes(childIterator.next(), graph);
-    }
-  }
-
   @NotNull
-  private static List<File> getOutputErlangModuleFiles(@NotNull List<String> erlangModulePathsToCompile,
+  private static List<File> getOutputErlangModuleFiles(@NotNull Collection<String> erlangModulePathsToCompile,
                                                        @NotNull final File outputDirectory) {
     return ContainerUtil.map(erlangModulePathsToCompile, new Function<String, File>() {
       @Override
       public File fun(String filePath) {
-        String name = FileUtil.getNameWithoutExtension(filePath);
-        return new File(outputDirectory.getAbsolutePath() + name + ".beam");
+        String name = FileUtil.getNameWithoutExtension(new File(filePath));
+        return new File(outputDirectory.getAbsolutePath() + '/' + name + ".beam");
```

Comment:
Avoid writing OS-dependent code. In this case, your code won't behave well on Windows.